source/Octopus.Tentacle/Services/Scripts/Kubernetes/KubernetesScriptServiceV1Alpha.cs

```diff
@@ -58,7 +58,16 @@ public async Task<KubernetesScriptStatusResponseV1Alpha> StartScriptAsync(StartK
                     return await GetResponse(trackedPod, 0, cancellationToken);
                 }
 
-                //TODO: consider adding an idempotent version of PrepareWorkspace
+                //Note:
+                //We shouldn't really need to worry about starting the same script twice,
+                //since we wait for the Pods to be reloaded from the K8s API on Tentacle restart.
+                //We might try to start the same script twice if
+                // - a Pod gets created
+                // - Tentacle restarts before returning
+                // - Server retries StartScriptAsync()
+                // - The Kubernetes API doesn't say that the Pod exists when we query it.
+                
+                //Note: PrepareWorkspace overwrites "command.Files", so it's not strictly idempotent
                 var workspace = await workspaceFactory.PrepareWorkspace(command.ScriptTicket,
```

Comment:
I think this is fine as it will be overwriting it with the same files anyway

Reply:
Just depends what happens if a script is trying to read the file as it's being overwritten